File: web/src/main/java/com/isystk/sample/web/base/controller/IDTypeEditor.java
package com.isystk.sample.web.base.controller;

import com.isystk.sample.common.dto.ID;
import java.beans.PropertyEditorSupport;

/**
 * 文字列をIDクラスに変換する
 */
public class IDTypeEditor extends PropertyEditorSupport {

  @Override
  public void setAsText(String text) throws IllegalArgumentException {
    try {
      Integer id = Integer.valueOf(text);
      setValue(ID.of(id));
    } catch (NumberFormatException e) {
      // nop
    }
  }
}

File: web/src/main/kotlin/com/isystk/sample/web/base/controller/IDTypeEditor.kt
package com.isystk.sample.web.base.controller

import com.isystk.sample.common.dto.ID
import java.beans.PropertyEditorSupport

/**
 * 文字列をIDクラスに変換する
 */
class IDTypeEditor : PropertyEditorSupport() {
    @Throws(IllegalArgumentException::class)
    override fun setAsText(text: String) {
        try {
            val id = Integer.valueOf(text)
            value = ID.of<Any>(id)
        } catch (e: NumberFormatException) {
            // nop
        }
    }
}